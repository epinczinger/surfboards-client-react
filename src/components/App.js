import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { getProducts, localStorageSignIn } from '../actions';
import SideBar from './SideBar';
import SignUp from './SignUp';
import SignIn from './SignIn';
import SurfboardDetails from './SurfboardDetails';
import AccesoryDetails from './AccesoryDetails';
import ItemList from '../containers/ItemList';
import FavouritesList from '../containers/FavouritesList';
import DefaultNotFound from '../containers/DefaultNotFound';
import Lifestyle from '../containers/Lifestyle';

const App = () => {
  const dispatch = useDispatch();
  const isLoggedIn = useSelector((state) => state.session[1]);

  useEffect(() => {
    dispatch(getProducts());
    if (window.localStorage.getItem('sessionID') && !isLoggedIn) {
      dispatch(localStorageSignIn(window.localStorage.getItem('sessionID')));
    }
  }, []);

  return (
    <div className="flex">
      <Router>
        <SideBar />
        <Switch>
          <Route path="/surfboards/:id" exact component={SurfboardDetails} />
          <Route path="/accesories/:id" exact component={AccesoryDetails} />
          <Route path="/sign_up" exact component={SignUp} />
          <Route path="/sign_in" exact component={SignIn} />
          <Route path="/surfboards" exact component={ItemList} />
          <Route path="/accesories" exact component={ItemList} />
          <Route path="/favourites" exact component={FavouritesList} />
          <Route path="/lifestyle" exact component={Lifestyle} />
          <Route path="*" component={DefaultNotFound} />
        </Switch>
      </Router>
    </div>
  );
};

export default App;
