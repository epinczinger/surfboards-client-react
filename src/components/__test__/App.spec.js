import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../reducers";

describe("The App component", () => {
  it("renders as expected matching snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
