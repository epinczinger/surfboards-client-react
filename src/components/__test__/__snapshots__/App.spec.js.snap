// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component renders as expected matching snapshot 1`] = `
<div
  className="flex"
>
  <div
    className="sm:flex sticky top-0 text-gray-800 flex-col justify-between md:w-56 bg-white h-screen border-r"
  >
    <div
      className="sm:hidden absolute top-1 left-1 flex justify-end text-2xl p-2 my-auto"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        â˜°
      </button>
    </div>
    <div
      className="hidden sm:block"
    >
      <div
        className="p-4 text-center hover:bg-blue-300 text-4xl hover:text-gray-100 font-semibold"
      >
        <a
          className="p-2"
          href="/"
          onClick={[Function]}
        >
          Surfshop
        </a>
      </div>
      <div
        className="flex justify-center text-sm text-gray-600 p-3"
      >
        <div
          className="hover:text-blue-400 transform hover:scale-110"
        >
          <a
            className="p-3"
            href="/sign_in"
            onClick={[Function]}
          >
            Sign In
          </a>
        </div>
        <div
          className="hover:text-blue-400 transform hover:scale-110"
        >
          <a
            className="p-3"
            href="/sign_up"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </div>
      </div>
    </div>
    <div
      className="text-left pl-3 pb-20 font-semibold text-2xl hidden sm:block"
    >
      <div
        className="hover:bg-blue-300 p-3 hover:text-gray-100"
      >
        <a
          className=""
          href="/surfboards"
          onClick={[Function]}
        >
          Surfboards
        </a>
      </div>
      <div
        className="hover:bg-blue-300 p-3 hover:text-gray-100"
      >
        <a
          className=""
          href="/accesories"
          onClick={[Function]}
        >
          Accesories
        </a>
      </div>
      <div
        className="hover:bg-blue-300 p-3 hover:text-gray-100"
      >
        <a
          className=""
          href="/lifestyle"
          onClick={[Function]}
        >
          Lifestyle
        </a>
      </div>
      <div
        className="hover:bg-blue-300 p-3 hover:text-gray-100"
      >
        <a
          className=""
          href="/favourites"
          onClick={[Function]}
        >
          Favourites
        </a>
      </div>
    </div>
    <div
      className="hidden sm:block"
    >
      <div
        className="p-4 flex text-lg items-center justify-around"
      >
        <a
          className="transition hover:text-blue-400 duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
          href="mailto:epinczinger@gmail.com"
        >
          <svg
            className="text-lg"
            fill="currentColor"
            height="21"
            version="1.1"
            viewBox="0 0 125 100"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M60,0 l50,0 a10,10 0 0,1 7,17 l-50,50 a10,10 0 0,1 -13,0 l-50,-50 a10,10 0 0,1 7,-17z"
            />
            <path
              d="M60,90 l54,0 a10,10 0 0,0 7,-7 l0,-60 -50,50 a15,15 0 0,1 -21,0 l-50,-50 0,60 a10,10 0 0,0 7,7z"
            />
          </svg>
        </a>
        <a
          className="transition hover:text-blue-400 duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
          href="https://www.github.com/epinczinger"
        >
          <svg
            fill="currentColor"
            height="21"
            viewBox="0 0 1792 1792"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M896 128q209 0 385.5 103t279.5 279.5 103 385.5q0 251-146.5 451.5t-378.5 277.5q-27 5-40-7t-13-30q0-3 .5-76.5t.5-134.5q0-97-52-142 57-6 102.5-18t94-39 81-66.5 53-105 20.5-150.5q0-119-79-206 37-91-8-204-28-9-81 11t-92 44l-38 24q-93-26-192-26t-192 26q-16-11-42.5-27t-83.5-38.5-85-13.5q-45 113-8 204-79 87-79 206 0 85 20.5 150t52.5 105 80.5 67 94 39 102.5 18q-39 36-49 103-21 10-45 15t-57 5-65.5-21.5-55.5-62.5q-19-32-48.5-52t-49.5-24l-20-3q-21 0-29 4.5t-5 11.5 9 14 13 12l7 5q22 10 43.5 38t31.5 51l10 23q13 38 44 61.5t67 30 69.5 7 55.5-3.5l23-4q0 38 .5 88.5t.5 54.5q0 18-13 30t-40 7q-232-77-378.5-277.5t-146.5-451.5q0-209 103-385.5t279.5-279.5 385.5-103zm-477 1103q3-7-7-12-10-3-13 2-3 7 7 12 9 6 13-2zm31 34q7-5-2-16-10-9-16-3-7 5 2 16 10 10 16 3zm30 45q9-7 0-19-8-13-17-6-9 5 0 18t17 7zm42 42q8-8-4-19-12-12-20-3-9 8 4 19 12 12 20 3zm57 25q3-11-13-16-15-4-19 7t13 15q15 6 19-6zm63 5q0-13-17-11-16 0-16 11 0 13 17 11 16 0 16-11zm58-10q-2-11-18-9-16 3-14 15t18 8 14-14z"
            />
          </svg>
        </a>
        <a
          className="transition hover:text-blue-400 duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
          href="https://www.linkedin.com/in/esteban-pinczinger"
        >
          <svg
            fill="currentColor"
            height="21"
            viewBox="0 0 1792 1792"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M477 625v991h-330v-991h330zm21-306q1 73-50.5 122t-135.5 49h-2q-82 0-132-49t-50-122q0-74 51.5-122.5t134.5-48.5 133 48.5 51 122.5zm1166 729v568h-329v-530q0-105-40.5-164.5t-126.5-59.5q-63 0-105.5 34.5t-63.5 85.5q-11 30-11 81v553h-329q2-399 2-647t-1-296l-1-48h329v144h-2q20-32 41-56t56.5-52 87-43.5 114.5-15.5q171 0 275 113.5t104 332.5z"
            />
          </svg>
        </a>
      </div>
      <div
        className="text-center text-sm pb-3 font-light"
      >
        <p>
           Created by 
        </p>
        <a
          className="hover:text-blue-500"
          href="https://www.github.com/epinczinger"
          rel="noreferrer"
          target="_blank"
        >
          Esteban Pinczinger
        </a>
      </div>
    </div>
  </div>
  <div
    className="w-fullbg-white font-semibold min-h-screen shadow-md py-5 sm:mx-auto overflow-hidden flex flex-col justify-around"
  >
    <div
      className="text-3xl text-gray-800 text-center"
    >
      <h1>
        Waves, waves waves.
      </h1>
    </div>
    <div
      className="sm:flex sm:my-2 justify-around"
    >
      <img
        alt="surfboard"
        className="himg w-50 object-cover"
        src="https://cdn.pixabay.com/photo/2016/03/09/15/16/wave-1246560_960_720.jpg"
      />
    </div>
    <div
      className="text-3xl text-gray-800 text-center"
    >
      <h1>
        Everyone wants to be there, but first you need some
        <a
          className="text-blue-500"
          href="/surfboards"
          onClick={[Function]}
        >
           
          Equipment
        </a>
      </h1>
    </div>
  </div>
</div>
`;
